File: Add Code Here/Java/QuickSort.java
import java.util.Scanner;

public class QuickSort {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int[] arr = new int[100];
        int n;
        System.out.print("Enter number of elements in the array: ");
        n = input.nextInt();
        System.out.println("Enter elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = input.nextInt();
        }
        System.out.println("\nThe unsorted array is:");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + "\t");
        }
        quickSort(arr, 0, n - 1);
        System.out.println("\nThe sorted array is:");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + "\t");
        }
    }

    public static int partition(int[] a, int beg, int end) {
        int left, right, temp, loc, flag;
        loc = left = beg;
        right = end;
        flag = 0;
        while (flag != 1) {
            while ((a[loc] <= a[right]) && (loc != right))
                right--;
            if (loc == right)
                flag = 1;
            else if (a[loc] > a[right]) {
                temp = a[loc];
                a[loc] = a[right];
                a[right] = temp;
                loc = right;
            }
            if (flag != 1) {
                while ((a[loc] >= a[left]) && (loc != left))
                    left++;
                if (loc == left)
                    flag = 1;
                else if (a[loc] < a[left]) {
                    temp = a[loc];
                    a[loc] = a[left];
                    a[left] = temp;
                    loc = left;
                }
            }
        }
        return loc;
    }

    public static void quickSort(int[] a, int beg, int end) {
        int loc;
        if (beg < end) {
            loc = partition(a, beg, end);
            quickSort(a, beg, loc - 1);
            quickSort(a, loc + 1, end);
        }
    }
}

File: Add Code Here/Python/quick sort.py
def partition(arr, beg, end):
    left, right, temp, loc, flag = 0, end, arr[beg], beg, 0
    while flag != 1:
        while (arr[loc] <= arr[right]) and (loc != right):
            right -= 1
        if loc == right:
            flag = 1
        else:
            temp = arr[loc]
            arr[loc], arr[right] = arr[right], arr[loc]
            loc = right
    while (arr[loc] >= arr[left]) and (loc != left):
        left += 1
    if loc == left:
        flag = 1
    else:
        temp = arr[loc]
        arr[loc], arr[left] = arr[left], arr[loc]
        loc = left
    return loc

def quick_sort(arr, beg, end):
    if beg < end:
        loc = partition(arr, beg, end)
        print("\nThe array after partition:")
        for i in range(beg, end + 1):
            print(arr[i], end=" ")
        print()
        quick_sort(arr, beg, loc - 1)
        print("\nThe array after left sub-array sorted is:")
        for i in range(beg, loc):
            print(arr[i], end=" ")
        print()
        quick_sort(arr, loc + 1, end)
        print("\nThe array after right sub-array sorted is:")
        for i in range(loc + 1, end + 1):
            print(arr[i], end=" ")
        print()

def main():
    n = int(input("Enter number of elements in the array: "))
    arr = [int(input(f"Enter element {i + 1}: ")) for i in range(n)]
    print("\nThe unsorted array is:")
    for i in range(n):
        print(arr[i], end=" ")
    quick_sort(arr, 0, n - 1)
    print("\nThe sorted array is:")
    for i in range(n):
        print(arr[i], end=" ")

if __name__ == "__main__":
    main()
